### PR TITLE
CPM-1003: Fix CategoryQueryHandler::byCodes() type error

### PR DESCRIPTION
**Pull request by jeremyBeaucousinAkeneo**
**Description (for Contributor and Core Developer)**

**Definition Of Done (for Core Developer only)**

- [ ] Tests
- [ ] Migration & Installer
- [ ] PM Validation (Story)
- [ ] Changelog (maintenance bug fixes)
- [ ] Tech Doc

### CODE REVIEW

**Review by mmetayer — COMMENTED** [review]
Does it also work for external API? (empty labels should be `{}`)

**Comment by jeremyBeaucousinAkeneo** [comment]
> 

Hello @mmetayer,

Yes It's also the case, and I has already been handled 😃.

**Review by mmetayer — APPROVED** [review]

**Review by antoinetrouve — APPROVED** [review]

**Review by pierallard — APPROVED** [review]
I don't like the usage of static functions as services, but it's not my bounded context :D
Enrichment part is OK for me.

File: src/Akeneo/Category/back/ServiceApi/InternalApiCategory.php [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+namespace Akeneo\Category\ServiceApi;
+
+use Akeneo\Category\Domain\Model\Enrichment\Category;
+use Akeneo\Category\Domain\ValueObject\ValueCollection;
+
+/**
+ * @copyright 2022 Akeneo SAS (https://www.akeneo.com)
+ * @license   http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ *
+ * @phpstan-import-type NormalizedValue from ValueCollection
```
[the remark this answers is not in the document]
Comment (jeremyBeaucousinAkeneo) [reply src/Akeneo/Category/back/ServiceApi/InternalApiCategory.php:14]: I'll clean that, I was implementing different solutions.

File: src/Akeneo/Category/back/ServiceApi/InternalApiCategory.php [hunk 2]
```diff
@@ -0,0 +1,103 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Akeneo\Category\ServiceApi;
```
Comment (pchasle) [inline src/Akeneo/Category/back/ServiceApi/InternalApiCategory.php:5]: This shouldn't be in the ServiceApi folder, but we can fix that later.

**Review by pchasle — APPROVED** [review]

**Pull request merged**